File: src/test/java/com/ject/studytrip/trip/helper/TripReportTestHelper.java
package com.ject.studytrip.trip.helper;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.trip.domain.model.TripReport;
import com.ject.studytrip.trip.domain.repository.TripReportRepository;
import com.ject.studytrip.trip.fixture.TripReportFixture;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class TripReportTestHelper {

    @Autowired private TripReportRepository tripReportRepository;

    public TripReport saveTripReport(Member member) {
        TripReport tripReport = TripReportFixture.createTripReport(member);
        return tripReportRepository.save(tripReport);
    }
}

File: src/test/kotlin/com/ject/studytrip/trip/helper/TripReportTestHelper.kt
package com.ject.studytrip.trip.helper

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.trip.domain.model.TripReport
import com.ject.studytrip.trip.domain.repository.TripReportRepository
import com.ject.studytrip.trip.fixture.TripReportFixture
import org.springframework.stereotype.Component

@Component
class TripReportTestHelper(
    private val tripReportRepository: TripReportRepository,
) {
    fun saveTripReport(member: Member): TripReport = tripReportRepository.save(TripReportFixture(member).create())

    fun saveDeletedTripReport(member: Member): TripReport =
        tripReportRepository.save(
            TripReportFixture(member).create().also {
                it.updateDeletedAt()
            },
        )
}
